business/workloads.go

```diff
@@ -1437,14 +1437,16 @@ func (in *WorkloadService) fetchWorkloadsFromCluster(ctx context.Context, cluste
 
 		if cnFound {
 			// Add the Proxy Status to the workload
+			isWaypoint := w.IsWaypoint()
 			for _, pod := range w.Pods {
-				isWaypoint := w.IsWaypoint()
 				if in.conf.ExternalServices.Istio.IstioAPIEnabled && (pod.HasIstioSidecar() || isWaypoint) {
 					pod.ProxyStatus = in.businessLayer.ProxyStatus.GetPodProxyStatus(cluster, namespace, pod.Name, !isWaypoint)
 				}
-				// Add the Proxy Status to the workload
+				// Add the Waypoint Workloads to the workload
 				if pod.AmbientEnabled() {
 					w.WaypointWorkloads = in.GetWaypointsForWorkload(ctx, *w, false)
+					// only need to do this work one time per workload
+					continue
```

Comment:
This continue wouldn't omit the rest of this iteration but it will continue next? so it might be more efficient to use a boolean flag? (Because checking the waypointworkloads len when it is really 0 can make this trigger again

Reply:
+1, this should be `break`.

Reply:
But in that case, I think the code to calculate the pod proxy status should be done in a separate loop; https://github.com/kiali/kiali/blob/master/business/workloads.go#L1442 (Otherwise must be a risk of not been calculated for all the pods)

Reply:
right again! I'll fix it.